Informatics II/001/Task 2/Patertuck Solution/Task2.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+int main() {
+    int dimension = 2;
+    int matrix[dimension*dimension];
+    int matrixsqr[dimension*dimension];
+    int i, j, column, row_start;
+
+    for (i=0; i<dimension * dimension; i++){
+        scanf("%d", &matrix[i]);
+    }
+    //calculating matrixsqr
+    for (i=0; i<dimension * dimension; i++){
+        column = i % dimension;
+        row_start = i - column;
+        matrixsqr[i] = 0;
+
+        for (j=0; j<dimension; j++){
+            matrixsqr[i] += matrix[row_start + j] * matrix[j * dimension + column];
```

Comment:
Interesting, does this work (row_start)?

Reply:
Yes! I actually looked over your code too and you are doing the exact same thing except that you divide your y by DIM and then multiplicate it with DIM again. Could be something you can improve in your code too :)

Reply:
Theorised that but was to lazy to go back to my code :) Nice! Yes, you right, would be an optimisation.